### PR TITLE
feat: Implement filtering by EnrollmentDate with offset in CPL [2.41-DHIS2-13795]

### PR DESCRIPTION
**Pull request by d-bernat**
Please see https://dhis2.atlassian.net/browse/DHIS2-13795

### CODE REVIEW

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java [hunk 1]
```diff
@@ -123,7 +123,7 @@ public JdbcEventAnalyticsTableManager( IdentifiableObjectManager idObjectManager
         new AnalyticsTableColumn( quote( "ao" ), CHARACTER_11, NOT_NULL, "ao.uid" ),
         new AnalyticsTableColumn( quote( "enrollmentdate" ), TIMESTAMP, "pi.enrollmentdate" ),
         new AnalyticsTableColumn( quote( "incidentdate" ), TIMESTAMP, "pi.incidentdate" ),
-        new AnalyticsTableColumn( quote( "executiondate" ), TIMESTAMP, "psi.executiondate" ),
+        new AnalyticsTableColumn( quote( "executiondate" ), TIMESTAMP, "psi.executiondate + INTERVAL '1 min' * 1439" ),
```
Comment (gnespolino) [inline dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java:126]: Why is this needed?

**Review by gnespolino — APPROVED** [review]
looks good to me, just a comment on a change I don't understand

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java (hunk 1 above)
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/table/JdbcEventAnalyticsTableManager.java]: done

File: dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/query/context/querybuilder/ContextUtils.java [hunk 2]
```diff
@@ -51,13 +51,16 @@
...
         return "select innermost_enr.*" +
             " from (select *," +
-            " row_number() over (partition by trackedentityinstanceuid order by enrollmentdate desc) as rn " +
+            " row_number() over (partition by trackedentityinstanceuid order by enrollmentdate "
+            + (offset > 0 ? "desc" : "asc") + ") as rn " +
             " from " + ANALYTICS_TEI_ENR + trackedEntityType.getUid().toLowerCase() +
             " where " + P_UID + " = " + sqlParameterManager.bindParamAndGetIndex( program.getElement().getUid() )
             + ") innermost_enr" +
-            " where innermost_enr.rn = 1";
+            " where innermost_enr.rn = " + Math.abs( offset );
```
[the remark this answers is not in the document]
Comment (d-bernat) [reply dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/tei/query/context/querybuilder/ContextUtils.java:63]: it is, but rn is row number (that's why 0 as offset is not allowed), to have whole picture: negative int mean ASC, positive DESC

**Review by maikelarabori — APPROVED** [review]

**Review by gnespolino — APPROVED** [review]
Looks great @d-bernat , great job!
